### PR TITLE
Validate quantization

### PR DESCRIPTION
**Pull request by yunfeng-scale**
Validate quantization values when creating endpoints

### CODE REVIEW

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 1]
```diff
@@ -196,10 +203,19 @@ def validate_num_shards(
...
+            f"Num shard {num_shards} must be the same as number of GPUs {gpus}."
+        )
+
+
+def validate_quantization(
+    quantize: Optional[Quantization], inference_framework: LLMInferenceFramework
+) -> None:
+    if quantize is not None and quantize not in _SUPPORTED_QUANTIZATIONS[inference_framework]:
+        raise ObjectHasInvalidValueException(
+            f"Quantization {quantize} is not supported for inference framework {inference_framework}."
```
Comment (yixu34) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:217]: Should we print out the list of valid quantization methods for the given inference framework as additional instruction to the user?

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 2]
```diff
@@ -196,10 +203,19 @@ def validate_num_shards(
     if inference_framework == LLMInferenceFramework.DEEPSPEED:
         if num_shards <= 1:
             raise ObjectHasInvalidValueException("DeepSpeed requires more than 1 GPU.")
-        if num_shards != gpus:
-            raise ObjectHasInvalidValueException(
-                f"DeepSpeed requires num shard {num_shards} to be the same as number of GPUs {gpus}."
-            )
+    if num_shards != gpus:
+        raise ObjectHasInvalidValueException(
+            f"Num shard {num_shards} must be the same as number of GPUs {gpus}."
```
Comment (yixu34) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:208]: Are there docstrings we need to update?

File: model-engine/model_engine_server/common/env_vars.py [hunk 3]
```diff
@@ -73,5 +74,5 @@ def get_boolean_env_var(name: str) -> bool:
     logger.warning("LOCAL development & testing mode is ON")
 
 GIT_TAG: str = os.environ.get("GIT_TAG", "GIT_TAG_NOT_FOUND")
-if GIT_TAG == "GIT_TAG_NOT_FOUND":
+if GIT_TAG == "GIT_TAG_NOT_FOUND" and "pytest" not in sys.modules:
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/common/env_vars.py:77]: make pytest work without specifying `GIT_TAG`

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 4]
```diff
@@ -136,6 +136,13 @@
     },
 }
 
+_SUPPORTED_QUANTIZATIONS: Dict[LLMInferenceFramework, List[Quantization]] = {
+    LLMInferenceFramework.DEEPSPEED: [],
+    LLMInferenceFramework.TEXT_GENERATION_INFERENCE: [Quantization.BITSANDBYTES],
+    LLMInferenceFramework.VLLM: [Quantization.AWQ],
+    LLMInferenceFramework.LIGHTLLM: [],
```
Comment (ian-scale) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:143]: probably best for a separate pr, but can you update the docs to specify which models in the model zoo support lightllm as inference framework?

**Review by ian-scale — APPROVED** [review]

**Pull request merged**